x-pack/legacy/plugins/index_lifecycle_management/public/services/ui_metric.js

```diff
@@ -29,9 +28,7 @@ import {
   defaultHotPhase,
 } from '../store/defaults';
 
-export function trackUiMetric(metricType) {
-  track(UIM_APP_NAME, metricType);
-}
+export const trackUiMetric = getUiStatsReporter(UIM_APP_NAME);
```

Comment:
I've found that interfaces that accept particular values are easy to use incorrectly, either due to typos ('count' vs 'conut') or to forgetting the value of the string you want to use (was it 'load' or 'loaded'?).

I think we can solve this in two ways:

1. Can the `ui_metric` plugin export constants that codify the acceptable values?
2. Can we export specific functions from this file which pre-bind calls to `trackUiMetric` with the various string values?

```js
import {
  getUiStatsReporter,
  METRIC_LOADED,
  METRIC_CLICK,
  METRIC_COUNT,
} from '../../../../../../src/legacy/core_plugins/ui_metric/public';

export const trackLoaded = eventName => trackUiMetric(METRIC_LOADED, eventName);
export const trackClick = eventName => trackUiMetric(METRIC_CLICK, eventName);
export const trackCount = eventName => trackUiMetric(METRIC_COUNT, eventName);
```

Reply:
I agree using an `enum` is a better approach, i'll update the code. It'll be imported from `@kbn/analytics` however not from `ui_metric`. Note that TS should fail on compile time, and the switch case for reporting will also through if someone miswrote the case.